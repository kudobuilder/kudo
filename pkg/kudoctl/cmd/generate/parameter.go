package generate

import (
	"github.com/spf13/afero"

	"github.com/kudobuilder/kudo/pkg/kudoctl/packages"
	"github.com/kudobuilder/kudo/pkg/kudoctl/packages/reader"
)

// AddParameter writes a parameter to the params.yaml file
func AddParameter(fs afero.Fs, path string, p *packages.Parameter) error {

	pf, err := reader.ReadDir(fs, path)
	if err != nil {
		return err
	}

	params := pf.Files.Params
	params.Parameters = append(params.Parameters, *p)

	return writeParameters(fs, path, *params)
}

func ParameterNameList(fs afero.Fs, path string) (paramNames []string, err error) {
	pf, err := reader.ReadDir(fs, path)
	if err != nil {
		return nil, err
	}

	for _, parameter := range pf.Files.Params.Parameters {
		paramNames = append(paramNames, parameter.Name)
	}

	return paramNames, nil
}
