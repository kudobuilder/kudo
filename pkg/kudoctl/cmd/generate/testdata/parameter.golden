apiVersion: kudo.dev/v1beta1
parameters:
- default: 1Gi
  description: Amount of memory to provide to Zookeeper pods
  name: memory
  required: true
- default: "0.25"
  description: Amount of cpu to provide to Zookeeper pods
  name: cpus
  required: true
- default: Bar
  name: Foo
